Foundation/FileHandle.swift

```diff
@@ -211,7 +211,46 @@ open class FileHandle : NSObject {
             fatalError("\(error)")
         }
     }
-    
+
+    private func _fstat() throws -> stat {
+        var statbuf = stat()
+        if fstat(_fd, &statbuf) < 0 {
+            throw _NSErrorWithErrno(errno, reading: true)
+        }
+        return statbuf
+    }
+
+    private func _fileSize() throws -> Int64 {
+        let statbuf = try _fstat()
+        return Int64(statbuf.st_size)
+    }
+
+    private var _isRegularFile: Bool? = nil
```

Comment:
Can this not change?  If I call the method and underneath remove the file and recreate it, the cached value will be wrong.  I don't think it is sufficient to just cache this value.

Reply:
If the file descriptor is already open, deleting/recreating the underlying file wont change what fd is pointing at.

Although I just noticed there is a `closeFile` method which should invalidate the cached values so I will update it for that.